src/s3/util/fileViewerManager.ts

```diff
@@ -2,30 +2,195 @@
  * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
  * SPDX-License-Identifier: Apache-2.0
  */
-
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { OutputChannel } from 'vscode'
 import { ext } from '../../shared/extensionGlobals'
-import { S3FileNode } from '../explorer/s3FileNode'
-import { Window } from '../../shared/vscode/window'
+import { makeTemporaryToolkitFolder } from '../../shared/filesystemUtilities'
 import { showOutputMessage } from '../../shared/utilities/messages'
-import { OutputChannel } from 'vscode'
+import { Commands } from '../../shared/vscode/commands'
+import { downloadWithProgress } from '../commands/downloadFileAs'
+import { S3FileNode } from '../explorer/s3FileNode'
+import { readablePath } from '../util'
+import { getStringHash } from '../../shared/utilities/textUtilities'
+
+const fs = require('fs')
+const SIZE_LIMIT = 4 * Math.pow(10, 6)
+
 export class S3FileViewerManager {
-    //private cache: Set<S3FileNode>
+    private cacheArn: Set<string>
     //private activeTabs: Set<S3Tab>
-    //private window: Window
+    private window: typeof vscode.window
     private outputChannel: OutputChannel
+    private commands: Commands
+    private tempLocation: string | undefined //TODOD: create temp
 
-    public constructor(window: Window = Window.vscode(), outputChannel = ext.outputChannel) {
-        //this.cache = new Set<S3FileNode>()
+    public constructor(
+        cacheArn: Set<string> = new Set<string>(),
+        window: typeof vscode.window = vscode.window,
+        commands = Commands.vscode(),
+        tempLocation?: string
+    ) {
+        this.cacheArn = cacheArn
         //this.activeTabs = new Set<S3Tab>()
-        //this.window = window
-        this.outputChannel = outputChannel
-        //this.createTemp()
-        showOutputMessage('initializing manager', outputChannel)
+        this.window = window
+        this.commands = commands
+        this.tempLocation = tempLocation
+        this.outputChannel = ext.outputChannel
     }
 
     public async openTab(fileNode: S3FileNode): Promise<void> {
-        showOutputMessage(`     manager initialized, file: ${fileNode.file.key}`, this.outputChannel)
+        showOutputMessage(
+            `++++++++++++++++++++++++++++++++manager was initialized, file: ${fileNode.file.key}`,
+            this.outputChannel
+        )
```

Comment:
change these debug things to `getLogger().debug()` calls. they are likely worth keeping around as log messages.